src/Arcus.WebApi.Security.Authentication/SharedAccessKey/SharedAccessKeyAuthenticationFilter.cs

```diff
@@ -20,21 +20,23 @@ namespace Arcus.WebApi.Security.Authentication.SharedAccessKey
     /// </remarks>
     public class SharedAccessKeyAuthenticationFilter : IAsyncAuthorizationFilter
     {
-        private readonly string _headerName, _secretName;
+        private readonly string _headerName, _queryParameterName, _secretName;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="SharedAccessKeyAuthenticationFilter"/> class.
         /// </summary>
         /// <param name="headerName">The name of the request header which value must match the stored secret.</param>
+        /// <param name="queryParameterName"></param>
         /// <param name="secretName">The name of the secret that's being retrieved using the <see cref="ISecretProvider.Get"/> call.</param>
         /// <exception cref="ArgumentException">When the <paramref name="headerName"/> is <c>null</c> or blank.</exception>
         /// <exception cref="ArgumentException">When the <paramref name="secretName"/> is <c>null</c> or blank.</exception>
-        public SharedAccessKeyAuthenticationFilter(string headerName, string secretName)
+        public SharedAccessKeyAuthenticationFilter(string headerName, string queryParameterName, string secretName)
         {
-            Guard.NotNullOrWhitespace(headerName, nameof(headerName), "Header name cannot be blank");
+            Guard.For<ArgumentException>(() => String.IsNullOrWhiteSpace(headerName) && String.IsNullOrWhiteSpace(queryParameterName), "Either Header name or Queryparameter name must be supplied.");
```

Comment:
Suggestion to fix weird casing
```suggestion
            Guard.For<ArgumentException>(() => String.IsNullOrWhiteSpace(headerName) && String.IsNullOrWhiteSpace(queryParameterName), "Either header name or query parameter name must be supplied.");
```